op-service/metrics/balance.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// 10^18
+var etherScalar = new(big.Int).Exp(big.NewInt(10), big.NewInt(18), nil)
+
+// weiToEther divides the wei value by 10^18 to get a number in ether as a float64
+func weiToEther(wei *big.Int) float64 {
+	num := new(big.Rat).SetInt(wei)
+	denom := new(big.Rat).SetInt(etherScalar)
+	num = num.Quo(num, denom)
+	f, _ := num.Float64()
+	return f
+}
+
+// LaunchBalanceMetrics fires off a go rountine that queries the balance of the supplied account & periodically records it
+// to the balance metric of the namespace. The balance of the account is recorded in Ether (not Wei).
+// Cancel the supplied context to shut down the go routine
+func LaunchBalanceMetrics(ctx context.Context, log log.Logger, r *prometheus.Registry, ns string, client *ethclient.Client, account common.Address) {
+	go func() {
+		balanceGuage := promauto.With(r).NewGauge(prometheus.GaugeOpts{
+			Namespace: ns,
+			Name:      "balance",
+			Help:      "balance (in ether) of account " + account.String(),
+		})
+
+		ticker := time.NewTicker(10 * time.Second)
```

Comment:
I feel like 10 seconds is a bit fast, I feel like 1 minute might be plenty, its just a lot of requests. The cost of running a node decreases the amount of users running nodes

Reply:
will do. I don't think we actually need that fine grain data b/c we can set alerts based on projected balances (using a linear prediction in grafana).